app/src/main/java/net/gsantner/markor/format/todotxt/TodoTxtTextActions.java

```diff
@@ -164,7 +170,21 @@ public void onClick(View view) {
                     return;
                 }
                 case R.string.tmaid_todotxt_current_date: {
-                    _hlEditor.getText().insert(origSelectionStart, SttCommander.getToday());
+                    String dueDateString = origTask.getDueDate();
+                    Calendar dueDate = parseDateString(dueDateString, Calendar.getInstance());
+
+                    DatePickerDialog.OnDateSetListener listener = (_view, year, month, day) -> {
+                        Calendar fmtCal = Calendar.getInstance();
+                        fmtCal.set(year, month, day);
+                        origTask.setDueDate(SttCommander.DATEF_YYYY_MM_DD.format(fmtCal.getTime()));
```

Comment:
@gsantner this doesn't work `sttcmd.regenerateTaskLine(updatedTask)` does not use the new due date

Reply:
As duedate is part of description (i.e. freetext that comes after creation date) it doesnt know where to put the duedate. Basically due is a "parsed only" value.

In that case its neccassary to do strreplace(description, olddue, newdue) && setDue(newdue) && regenerate

Reply:
I see. I am probably going to switch to using the regexReplaceAction then.